File: app/src/main/java/com/github/oryanmat/trellowidget/util/Json.java
package com.github.oryanmat.trellowidget.util;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

public class Json {
    static Gson gson = new Gson();

    private Json() {
    }

    public static Gson get() {
        if (gson == null) {
            gson = new Gson();
        }

        return gson;
    }

    public static <T> T tryParseJson(String json, Class<T> c, T defaultValue) {
        try {
            return gson.fromJson(json, c);
        } catch (JsonSyntaxException e) {
            return defaultValue;
        }
    }
}

File: app/src/main/kotlin/com/github/oryanmat/trellowidget/util/Json.kt
package com.github.oryanmat.trellowidget.util

import com.google.gson.Gson
import com.google.gson.JsonSyntaxException

import java.lang.reflect.Type

object Json {
    private val gson = Gson()

    fun toJson(src: Any): String = gson.toJson(src)

    fun <T> fromJson(json: String, c: Class<T>): T = gson.fromJson(json, c)

    fun <T> tryParseJson(json: String, c: Class<T>, defaultValue: T): T = try {
        gson.fromJson(json, c)
    } catch (e: JsonSyntaxException) {
        defaultValue
    }

    fun <T> tryParseJson(json: String, type: Type, defaultValue: T): T = try {
        gson.fromJson<T>(json, type)
    } catch (e: JsonSyntaxException) {
        defaultValue
    }
}